soccer/gameplay/skills/pivot_kick.py

```diff
@@ -71,6 +82,24 @@ def __init__(self):
         self.dribbler_power = constants.Robot.Dribbler.StandardPower
         self.aim_params = {'desperate_timeout': float("inf")}
 
+    def facing_opp_goal(self):
+        robot = self.subbehavior_with_name('aim').robot
+        if robot is None:
+            return False
+        robotAngle = robot.angle
+        a1 = math.atan(abs(robot.pos.y - constants.Field.Length) / (robot.pos.x - constants.Field.GoalWidth))
+        if a1 < 0:
+            a1 +=  math.pi
+        a2 = math.atan(abs(robot.pos.y - constants.Field.Length) / (robot.pos.x + constants.Field.GoalWidth))
+        if a2 < 0:
+            a2 += math.pi
+        if robotAngle > min(a1, a2) and robotAngle < max(a1, a2) and robot.pos.y > constants.Field.Length / 2:
+            print('EARLY KIck')
+            main.system_state().draw_text('Early kick', robot.pos, 'PivotKick')
+            return True
+        return False
+
+
```

Comment:
Did we ever test this to make sure it works correctly?

Reply:
We should also probs use our robocup.Point lib instead of atan for consistency/simplicity.

Reply:
I never got a chance to accurately test this on real robots

Reply:
This looks good in sim, I'll see if I can test it on the real boys